scripts/build_cop_profiles/BaseCopApproximator.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+# SPDX-FileCopyrightText: : 2020-2024 The PyPSA-Eur Authors
+#
+# SPDX-License-Identifier: MIT
+
+from abc import ABC, abstractmethod
+from typing import Union
+
+import numpy as np
+import xarray as xr
+
+
+class BaseCopApproximator(ABC):
+    """
+    Abstract class for approximating the coefficient of performance (COP) of a
+    heat pump.
+    """
+
+    def __init__(
+        self,
+        forward_temperature_celsius: Union[xr.DataArray, np.array],
+        source_inlet_temperature_celsius: Union[xr.DataArray, np.array],
+    ):
+        """
+        Initialize CopApproximator.
+
+        Parameters:
+        ----------
+        forward_temperature_celsius : Union[xr.DataArray, np.array]
+            The forward temperature in Celsius.
+        return_temperature_celsius : Union[xr.DataArray, np.array]
+            The return temperature in Celsius.
+        """
+        pass
+
+    @abstractmethod
+    def approximate_cop(self) -> Union[xr.DataArray, np.array]:
+        """
+        Approximate heat pump coefficient of performance (COP).
+
+        Returns:
+        -------
+        Union[xr.DataArray, np.array]
+            The calculated COP values.
+        """
+        pass
+
+    def celsius_to_kelvin(
```

Comment:
Is it necessary to duplicate the functions, once with `@staticmethod` decorator?

Reply:
It very much isn't.